lib/spack/spack/test/concretize_requirements.py

```diff
@@ -248,6 +248,41 @@ def test_requirement_adds_multiple_new_versions(
     assert s2.satisfies("@2.3")
 
 
+def test_preference_adds_new_version(
+    concretize_scope, test_repo, mock_git_version_info, monkeypatch
+):
+    if spack.config.get("config:concretizer") == "original":
+        pytest.skip("Original concretizer does not support configuration" " requirements")
+
+    repo_path, filename, commits = mock_git_version_info
+    monkeypatch.setattr(
+        spack.package_base.PackageBase, "git", "file://%s" % repo_path, raising=False
```

Comment:
Notice that you can't concat `file://` with a path in general (like windows), see `spack.util.url.path_to_file_url(...)`

Reply:
Ping